app/src/main/java/com/example/songme/ui/main/MainActivity.kt

```diff
@@ -2,11 +2,55 @@ package com.example.songme.ui.main
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.MenuItem
+import androidx.fragment.app.Fragment
 import com.example.songme.R
+import com.example.songme.ui.genres.GenresFragment
+import com.example.songme.ui.home.ActionBarFragment
+import com.example.songme.ui.home.HomeFragment
+import com.example.songme.ui.mymusic.MyMusicFragment
+import com.google.android.material.bottomnavigation.BottomNavigationView
+import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        initView()
+        initListener()
+    }
+
+    private fun initView() {
+        addFragment(ActionBarFragment())
+        replaceFragment(HomeFragment())
+    }
+
+    private fun initListener() {
+        bottomNav?.setOnNavigationItemSelectedListener(this)
+        bottomNav.menu.findItem(R.id.navigation_home).isChecked = true
+    }
+
+    private fun addFragment(fragment: Fragment) {
+        supportFragmentManager.beginTransaction()
+            .add(R.id.frameActionBar, fragment)
+            .addToBackStack(null)
```

Comment:
Em thử bỏ dòng này chạy xem có gì thay đổi không

Reply:
khi e bỏ dòng addToBackStack(null) đi thì thay đổi nằm ở chỗ khi ta replace thì 1 fragment sau sẽ thay thế vào vị trí của fragment đầu tiên. Tuy fragment đầu đã gọi onPause, onStop, onDestroyView nhưng nó chưa bị destroy. Nó được put vào backStack và ta hoàn toàn có thể nhấn back button để pop nó ra khỏi backStack và trở lại. Còn khi không sử dụng addToBackStack thì back button chỉ có tác dụng với activity, như trong ứng dụng của e là sẽ out khỏi ứng dụng.